vyper/parser/stmt.py

```diff
@@ -546,16 +568,39 @@ def parse_for(self):
 
             # Type 1 for, e.g. for i in range(10): ...
             if num_of_args == 1:
-                arg0_val = self._get_range_const_value(arg0)
-                start = LLLnode.from_list(0, typ='int128', pos=getpos(self.stmt))
+                arg0_expr = self._get_range_const_value_expr(arg0)
+                arg0_val = arg0_expr.value
+                out_type_str = arg0_expr.typ.typ
```

Comment:
how about
```suggestion
                out_type = arg0_expr.typ
                start = LLLnode.from_list(0, typ=out_type, pos=getpos(self.stmt))
```

Reply:
Yeah; that didn't work... we need the string type later.